cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -41,6 +42,14 @@ func (e *ec2) InstanceIfExists(id *string) (*ec2svc.Instance, error) {
 	return nil, nil
 }
 
+func (e *ec2) TerminateInstance(instanceID *string) error {
```

Comment:
this ec2 object needs to be replaced by the ec2 mock we're using like this: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cloud/aws/services/ec2/subnets_test.go#L31

But I'm ok if you want to leave as is. We can do this work later.

Reply:
This isn't a direct EC2 call, but a wrapper for the real `ec2.TerminateInstances` in the same way that `CreateInstance` wraps `ec2.RunInstances`. I'm not sure if that changes what needs to be done here.

I think for this work right now, I'd be happier having this get merged and then working on improving the testing in a separate PR, as I think it will take me a little time to get up to speed on proper mocking.